### PR TITLE
Allow configurable Http KeepAlive parameters

### PR DESCRIPTION
**Pull request by Paciv**

### CODE REVIEW

File: src/KubernetesClient/Kubernetes.ConfigInit.cs [hunk 1]
```diff
@@ -117,9 +117,9 @@ private void CreateHttpClient(DelegatingHandler[] handlers, KubernetesClientConf
 #if NET5_0_OR_GREATER
             FirstMessageHandler = HttpClientHandler = new SocketsHttpHandler
```
Comment (tg123) [inline src/KubernetesClient/Kubernetes.ConfigInit.cs:118]: SocketsHttpHandler has so many props
why not create a `callback/factory` and you can put socket callback there as well

Comment (Paciv) [reply src/KubernetesClient/Kubernetes.ConfigInit.cs]: Done

File: src/KubernetesClient/KubernetesClientConfiguration.cs [hunk 2]
```diff
@@ -89,5 +90,19 @@ public partial class KubernetesClientConfiguration
...
         /// <value>timeout</value>
         public TimeSpan HttpClientTimeout { get; set; } = TimeSpan.FromSeconds(100);
+
+#if NET5_0_OR_GREATER
+
+        /// <summary>
+        ///     Gets or sets the SocketsHttpHandler factory.
+        /// </summary>
+        /// <value>The SocketsHttpHandler factory.</value>
+        public Func<SocketsHttpHandler> SocketsHttpHandlerFactory { get; set; } = () => new SocketsHttpHandler
```
Comment (tg123) [inline src/KubernetesClient/KubernetesClientConfiguration.cs:100]:
```suggestion
        public Func<SocketsHttpHandler> FirstMessageHandlerFactory { get; set; } = () => new SocketsHttpHandler
```

File: src/KubernetesClient/Kubernetes.ConfigInit.cs [hunk 3]
```diff
@@ -115,12 +115,7 @@ private void InitializeFromConfig(KubernetesClientConfiguration config)
...
-            FirstMessageHandler = HttpClientHandler = new SocketsHttpHandler
-            {
-                KeepAlivePingPolicy = HttpKeepAlivePingPolicy.WithActiveRequests,
-                KeepAlivePingDelay = TimeSpan.FromMinutes(3),
-                KeepAlivePingTimeout = TimeSpan.FromSeconds(30),
-            };
+            FirstMessageHandler = HttpClientHandler = config.SocketsHttpHandlerFactory();
 
             HttpClientHandler.SslOptions.ClientCertificates = new X509Certificate2Collection();
 #else
```
Comment (tg123) [inline src/KubernetesClient/Kubernetes.ConfigInit.cs:121]: could you please also add similar factory to `#else` (netstandard2)

**Review by tg123 — COMMENTED** [review]
thanks for the PR
could you please sign CLA?

File: src/KubernetesClient/Kubernetes.ConfigInit.cs (hunk 3 above)
Comment (Paciv) [reply src/KubernetesClient/Kubernetes.ConfigInit.cs:121]: Ok, but I'll switch to a callback instead, I don't like seeing too much code on the config side

**Comment by tg123** [comment]
Thanks and welcome
/LGTM

**Review by tg123 — APPROVED** [review]

**Pull request merged**